### ISSUE TITLE
环境问题

### ISSUE THREAD

**Issue by plyon1** [xhonghu/CESMM#1]
您好，请问可以告诉我在构建这个工程的时候你的环境是什么样的，包括python，torch，cuda版本等，我在配置环境的时候一直出错
Disabling PyTorch because PyTorch >= 2.1 is required but found 1.13.0+cu117

**Comment by xhonghu** [comment]
抱歉最近在忙投稿没有及时看到issue。是的这个环境不太好配，主要是CTC包版本限制比较多。我提供一个conda环境的打包版本的链接在 https://huggingface.co/datasets/xhonghu/Environment/tree/main
具体配置如下：
# packages in environment at /home/hhx22/conda_env/SLRT:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                  main.conda
_openmp_mutex             5.1                 1_gnu.conda
absl-py                   2.1.0                     <pip>
adaPool                   0.1                       <pip>
aiohappyeyeballs          2.4.3                     <pip>
aiohttp                   3.11.7                    <pip>
aiosignal                 1.3.1                     <pip>
async-timeout             5.0.1                     <pip>
attrs                     24.2.0                    <pip>
blas                      1.0                   mkl.conda
Brotli                    1.0.9                     <pip>
brotli-python             1.0.9           py310h6a678d5_8.conda
bzip2                     1.0.8           h5eee18b_6.conda
ca-certificates           2024.9.24       h06a4308_0.conda
cachetools                5.5.0                     <pip>
certifi                   2024.8.30       py310h06a4308_0.conda
certifi                   2024.8.30                 <pip>
charset-normalizer        3.3.2           pyhd3eb1b0_0.conda
charset-normalizer        3.3.2                     <pip>
click                     8.1.7                     <pip>
conda-pack                0.8.1                     <pip>
ctcdecode                 1.0.3                     <pip>
cudatoolkit               11.3.1          h2bc3f7f_2.conda
cycler                    0.12.1                    <pip>
docker-pycreds            0.4.0                     <pip>
einops                    0.8.0                     <pip>
ffmpeg                    4.3                  hf484d3e_0    pytorch
filelock                  3.16.1                    <pip>
freetype                  2.12.1          h4a9f257_0.conda
frozenlist                1.5.0                     <pip>
fsspec                    2024.10.0                 <pip>
fvcore                    0.1.5.post20221221           <pip>
gitdb                     4.0.11                    <pip>
GitPython                 3.1.43                    <pip>
gmp                       6.2.1           h295c915_3.conda
gnutls                    3.6.15          he1e5248_0.conda
grpcio                    1.68.0                    <pip>
huggingface-hub           0.26.2                    <pip>
idna                      3.7                       <pip>
idna                      3.7             py310h06a4308_0.conda
intel-openmp              2023.1.0        hdb19cb5_46306.conda
iopath                    0.1.10                    <pip>
Jinja2                    3.1.4                     <pip>
joblib                    1.4.2                     <pip>
jpeg                      9e              h5eee18b_3.conda
kiwisolver                1.4.7                     <pip>
lame                      3.100           h7b6447c_0.conda
lcms2                     2.12            h3be6417_0.conda
ld_impl_linux-64          2.40            h12ee557_0.conda
lerc                      3.0             h295c915_0.conda
libdeflate                1.17            h5eee18b_1.conda
libffi                    3.4.4           h6a678d5_1.conda
libgcc-ng                 11.2.0          h1234567_1.conda
libgomp                   11.2.0          h1234567_1.conda
libiconv                  1.16            h5eee18b_3.conda
libidn2                   2.3.4           h5eee18b_0.conda
libpng                    1.6.39          h5eee18b_0.conda
libstdcxx-ng              11.2.0          h1234567_1.conda
libtasn1                  4.19.0          h5eee18b_0.conda
libtiff                   4.5.1           h6a678d5_0.conda
libunistring              0.9.10          h27cfd23_0.conda
libuuid                   1.41.5          h5eee18b_0.conda
libwebp-base              1.3.2           h5eee18b_1.conda
lz4-c                     1.9.4           h6a678d5_1.conda
Markdown                  3.7                       <pip>
MarkupSafe                3.0.2                     <pip>
matplotlib                3.4.3                     <pip>
mkl                       2023.1.0        h213fc3f_46344.conda
mkl-service               2.4.0           py310h5eee18b_1.conda
mkl-service               2.4.0                     <pip>
mkl_fft                   1.3.11                    <pip>
mkl_fft                   1.3.11          py310h5eee18b_0.conda
mkl_random                1.2.8                     <pip>
mkl_random                1.2.8           py310h1128e8f_0.conda
multidict                 6.1.0                     <pip>
ncurses                   6.4             h6a678d5_0.conda
nettle                    3.7.3           hbbd107a_1.conda
numpy                     2.0.1           py310h5f9d8c6_1.conda
numpy                     1.21.2                    <pip>
nvidia-ml-py              12.560.30                 <pip>
nvitop                    1.4.0                     <pip>
opencv-python             4.5.5.64                  <pip>
openh264                  2.1.1           h4ff587b_0.conda
openjpeg                  2.5.2           he7f1fd0_0.conda
openssl                   3.0.15          h5eee18b_0.conda
packaging                 24.2                      <pip>
pandas                    1.3.4                     <pip>
Pillow                    9.4.0                     <pip>
pip                       24.2            py310h06a4308_0.conda
pip                       24.2                      <pip>
platformdirs              4.3.6                     <pip>
portalocker               3.0.0                     <pip>
propcache                 0.2.0                     <pip>
protobuf                  5.28.3                    <pip>
psutil                    6.1.0                     <pip>
pyparsing                 3.2.0                     <pip>
PySocks                   1.7.1                     <pip>
pysocks                   1.7.1           py310h06a4308_0.conda
python                    3.10.14         h955ad1f_1.conda
python-dateutil           2.9.0.post0               <pip>
pytorch                   1.12.0          py3.10_cuda11.3_cudnn8.3.2_0    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytorch-wavelets          1.3.0                     <pip>
pytz                      2024.2                    <pip>
PyWavelets                1.4.1                     <pip>
PyYAML                    6.0                       <pip>
readline                  8.2             h5eee18b_0.conda
regex                     2024.11.6                 <pip>
requests                  2.32.3          py310h06a4308_1.conda
requests                  2.32.3                    <pip>
safetensors               0.4.5                     <pip>
scikit-learn              1.5.2                     <pip>
scipy                     1.7.2                     <pip>
sentencepiece             0.2.0                     <pip>
sentry-sdk                2.19.0                    <pip>
setproctitle              1.3.4                     <pip>
setuptools                75.1.0                    <pip>
setuptools                75.1.0          py310h06a4308_0.conda
six                       1.16.0                    <pip>
smmap                     5.0.1                     <pip>
sqlite                    3.45.3          h5eee18b_0.conda
tabulate                  0.9.0                     <pip>
tbb                       2021.8.0        hdb19cb5_0.conda
tensorboard               2.18.0                    <pip>
tensorboard-data-server   0.7.2                     <pip>
termcolor                 2.5.0                     <pip>
thop-0.1.1                2209072238                <pip>
threadpoolctl             3.5.0                     <pip>
timm                      1.0.7                     <pip>
tk                        8.6.14          h39e8969_0.conda
tokenizers                0.20.3                    <pip>
torch                     1.12.0                    <pip>
torch_geometric           2.5.2                     <pip>
torchaudio                0.12.0              py310_cu113    pytorch
torchaudio                0.12.0                    <pip>
torchvision               0.13.0                    <pip>
torchvision               0.13.0              py310_cu113    pytorch
tqdm                      4.62.3                    <pip>
transformers              4.46.3                    <pip>
typing_extensions         4.11.0          py310h06a4308_0.conda
typing_extensions         4.11.0                    <pip>
tzdata                    2024b           h04d1e81_0.conda
urllib3                   2.2.3           py310h06a4308_0.conda
urllib3                   2.2.3                     <pip>
wandb                     0.17.4                    <pip>
Werkzeug                  3.1.3                     <pip>
wheel                     0.44.0                    <pip>
wheel                     0.44.0          py310h06a4308_0.conda
xz                        5.4.6           h5eee18b_1.conda
yacs                      0.1.8                     <pip>
yarl                      1.18.0                    <pip>
zlib                      1.2.13          h5eee18b_1.conda
zstd                      1.5.6           hc292b87_0.conda
